python/lsst/ts/cbp/component.py

```diff
@@ -179,7 +183,7 @@ def get_azimuth(self):
         -------
 
         """
-        self.socket.sendall("az=?\r".encode('ascii'))
+        self.socket.sendall("az=?\r".encode("ascii"))
```

Comment:
Please do your encoding and decoding in one place -- e.g. a method that sends data.

Reply:
Also if there is any risk that two sections of code may be sending commands and awaiting replies at the same time (which is very likely if you want the system to be responsive), then it is much, much safer to use a method to both send the command and wait for the reply or replies, and have a resource lock.

This method should also append the \r -- so sending a command just requires formatting the command itself, none of the extra stuff. That'll make the code a lot easier to read and avoids dangerous repetition.

Reply:
Since this is part of a salobj-based CSC, shouldn't you use asyncio-based sockets? Please don't mix thread-based code and asyncio code if you can avoid it. ts_ATDome is a good simple example of code that opens a socket and communicates with it. In particular the run_command methods in ATDomeCsc for this bit.

Reply:
Filed ticket for asyncio socket support.

Reply:
Done in the PR.